running job: 1
WARNING:__main__:
*****************************************
Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
*****************************************
Starting... loading imports
Starting... loading imports
2023-02-19 19:22:45
2023-02-19 19:22:45
2023-02-19 19:25:49 Loaded imports!
2023-02-19 19:25:49 Loaded imports!
Using cache found in /home-mscluster/erex/.cache/torch/hub/facebookresearch_xcit_main
Using cache found in /home-mscluster/erex/.cache/torch/hub/facebookresearch_xcit_main
| distributed init (rank 0): env://
| distributed init (rank 1): env://
/home-mscluster/erex/anaconda3/lib/python3.9/site-packages/torchvision/transforms/transforms.py:891: UserWarning: Argument 'interpolation' of type int is deprecated since 0.13 and will be removed in 0.15. Please use InterpolationMode enum.
  warnings.warn(
git:
  sha: cb711401860da580817918b9167ed73e3eef3dcf, status: has uncommited changes, branch: main

arch: vit_small
batch_size_per_gpu: 64
clip_grad: 3.0
data_path: /home-mscluster/erex/research_project/datasets/NLST_100k
dist_url: env://
drop_path_rate: 0.1
epochs: 500
freeze_last_layer: 1
global_crops_scale: (0.4, 1.0)
gpu: 0
local_crops_number: 8
local_crops_scale: (0.05, 0.4)
local_rank: 0
lr: 0.0005
min_lr: 1e-06
momentum_teacher: 0.996
norm_last_layer: True
num_workers: 10
optimizer: adamw
out_dim: 65536
output_dir: /home-mscluster/erex/research_project/models/dino/experiments/pre-trainingViT100k/saving_dir1
patch_size: 16
rank: 0
saveckp_freq: 20
seed: 0
teacher_temp: 0.04
use_bn_in_head: False
use_fp16: True
warmup_epochs: 10
warmup_teacher_temp: 0.04
warmup_teacher_temp_epochs: 0
weight_decay: 0.04
weight_decay_end: 0.4
world_size: 2
2023-02-19 19:26:28 Preparing data ...
/home-mscluster/erex/anaconda3/lib/python3.9/site-packages/torchvision/transforms/transforms.py:891: UserWarning: Argument 'interpolation' of type int is deprecated since 0.13 and will be removed in 0.15. Please use InterpolationMode enum.
  warnings.warn(
Traceback (most recent call last):
  File "/home-mscluster/erex/research_project/models/dino/main_dino.py", line 486, in <module>
Traceback (most recent call last):
  File "/home-mscluster/erex/research_project/models/dino/main_dino.py", line 486, in <module>
    train_dino(args)
  File "/home-mscluster/erex/research_project/models/dino/main_dino.py", line 157, in train_dino
    dataset = CachedImageFolder(args.data_path, transform=transform)
  File "/home-mscluster/erex/anaconda3/lib/python3.9/site-packages/torchvision/datasets/folder.py", line 310, in __init__
    train_dino(args)
  File "/home-mscluster/erex/research_project/models/dino/main_dino.py", line 157, in train_dino
    dataset = CachedImageFolder(args.data_path, transform=transform)
  File "/home-mscluster/erex/anaconda3/lib/python3.9/site-packages/torchvision/datasets/folder.py", line 310, in __init__
    super().__init__(
  File "/home-mscluster/erex/anaconda3/lib/python3.9/site-packages/torchvision/datasets/folder.py", line 145, in __init__
    super().__init__(
  File "/home-mscluster/erex/anaconda3/lib/python3.9/site-packages/torchvision/datasets/folder.py", line 145, in __init__
    classes, class_to_idx = self.find_classes(self.root)
  File "/home-mscluster/erex/research_project/datasets/cache_filenames.py", line 17, in decorated
    classes, class_to_idx = self.find_classes(self.root)
  File "/home-mscluster/erex/research_project/datasets/cache_filenames.py", line 17, in decorated
    out = f(self, directory, *args, **kwargs)
  File "/home-mscluster/erex/research_project/datasets/cache_filenames.py", line 26, in find_classes
    classes = super().find_classes(directory, *args, **kwargs)
  File "/home-mscluster/erex/anaconda3/lib/python3.9/site-packages/torchvision/datasets/folder.py", line 219, in find_classes
    return find_classes(directory)
  File "/home-mscluster/erex/anaconda3/lib/python3.9/site-packages/torchvision/datasets/folder.py", line 43, in find_classes
    raise FileNotFoundError(f"Couldn't find any class folder in {directory}.")
FileNotFoundError: Couldn't find any class folder in /home-mscluster/erex/research_project/datasets/NLST_100k.
    out = f(self, directory, *args, **kwargs)
  File "/home-mscluster/erex/research_project/datasets/cache_filenames.py", line 26, in find_classes
    classes = super().find_classes(directory, *args, **kwargs)
  File "/home-mscluster/erex/anaconda3/lib/python3.9/site-packages/torchvision/datasets/folder.py", line 219, in find_classes
    return find_classes(directory)
  File "/home-mscluster/erex/anaconda3/lib/python3.9/site-packages/torchvision/datasets/folder.py", line 43, in find_classes
    raise FileNotFoundError(f"Couldn't find any class folder in {directory}.")
FileNotFoundError: Couldn't find any class folder in /home-mscluster/erex/research_project/datasets/NLST_100k.
ERROR:torch.distributed.elastic.multiprocessing.api:failed (exitcode: 1) local_rank: 0 (pid: 2182951) of binary: /home-mscluster/erex/anaconda3/bin/python3
Traceback (most recent call last):
  File "/home-mscluster/erex/anaconda3/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home-mscluster/erex/anaconda3/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home-mscluster/erex/anaconda3/lib/python3.9/site-packages/torch/distributed/run.py", line 765, in <module>
    main()
  File "/home-mscluster/erex/anaconda3/lib/python3.9/site-packages/torch/distributed/elastic/multiprocessing/errors/__init__.py", line 345, in wrapper
    return f(*args, **kwargs)
  File "/home-mscluster/erex/anaconda3/lib/python3.9/site-packages/torch/distributed/run.py", line 761, in main
    run(args)
  File "/home-mscluster/erex/anaconda3/lib/python3.9/site-packages/torch/distributed/run.py", line 752, in run
    elastic_launch(
  File "/home-mscluster/erex/anaconda3/lib/python3.9/site-packages/torch/distributed/launcher/api.py", line 131, in __call__
    return launch_agent(self._config, self._entrypoint, list(args))
  File "/home-mscluster/erex/anaconda3/lib/python3.9/site-packages/torch/distributed/launcher/api.py", line 245, in launch_agent
    raise ChildFailedError(
torch.distributed.elastic.multiprocessing.errors.ChildFailedError: 
============================================================
main_dino.py FAILED
------------------------------------------------------------
Failures:
[1]:
  time      : 2023-02-19_19:30:31
  host      : mscluster108.ms.wits.ac.za
  rank      : 1 (local_rank: 1)
  exitcode  : 1 (pid: 2182952)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
------------------------------------------------------------
Root Cause (first observed failure):
[0]:
  time      : 2023-02-19_19:30:31
  host      : mscluster108.ms.wits.ac.za
  rank      : 0 (local_rank: 0)
  exitcode  : 1 (pid: 2182951)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
============================================================
